### PR TITLE
Tasks till Day 8

### PR DESCRIPTION
**Pull request by Rahul-Successive**

### CODE REVIEW

File: extra/utils/helpers.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+
+export default  function hasPermission(moduleName, role,permissionType,permission){
+   
+    if(Object.keys(permission)[0]  == moduleName){
+
+        if(permission[moduleName].hasOwnProperty(permissionType)){
+
+            if((permissionType == 'all' && permission[moduleName].all[0] == role)  ||
```
Comment (Rahul-Successive) [inline extra/utils/helpers.js]: use === instead of ==
have you setup tslint?

File: extraTs/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    }
+
+    let validUsers=0, invalidUsers=0
+
+    let a
+    let b
+   export function validateUsers(users:any){
+
+    for(let i=0;i<users.length;i++){
```
Comment (Rahul-Successive) [inline extraTs/utils/helpers.ts:14]: use forEach instead of for

File: src/Server.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        catch(error){
+            throw error;
+            
+        }
+    }
+
+    run(){
+        try{
+            this.app = express()
```
Comment (Rahul-Successive) [inline src/Server.ts:47]: don't you think it should be in the constructor?

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+let config:EnvironmentVariables = Object.freeze({
+    PORT : process.env.PORT,
+    NODE_ENV : process.env.NODE_ENV
+})
+
+console.log(process.env.PORT)
+let serverDao = new ServerDao(config)
+
+let run =serverDao.run()
```
Comment (Rahul-Successive) [inline src/index.ts]: use method chaining over here

File: src/libs/routes/notFoundRoute.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+
+this.app.use(function (err, req, res, next) {
```
Comment (Rahul-Successive) [inline src/libs/routes/notFoundRoute.ts]: what is 'this' over here?

File: src/Server.ts (hunk 3 above)
Comment (pankajchibhrani-successive) [reply src/Server.ts:47]: i did not get you as i am calling run method from **index.ts**

**Pull request merged**